### PR TITLE
feat(Accordion): adding support for nested accordion

### PR DESCRIPTION
**Pull request by priyankar205**
Adding support for nested accordion. Also added an example.

### TODO

- [ ] Conformance test
- [ ] Minimal doc site example
- [ ] Stardust base theme
- [ ] Teams Light theme
- [ ] Teams Dark theme
- [ ] Teams Contrast theme
- [ ] Confirm RTL usage
- [ ] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [ ] Update the CHANGELOG.md

### CODE REVIEW

File: src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -170,22 +178,28 @@ class Accordion extends AutoControlledComponent<Extendable<IAccordionProps>, any
...
       children.push(
         AccordionContent.create(content, {
           defaultProps: { active },
+          generateKey: true,
           render: renderContent,
         }),
       )
+
+      if (active && accordion) {
+        children.push(Accordion.create(accordion))
```
Comment (gopalgoel19) [inline src/components/Accordion/Accordion.tsx:201]: Maybe give ```marginLeft``` style to the nested accordion to have a better visual representation.

```jsx
children.push(Accordion.create(accordion, {
    defaultProps: { styles: {
        marginLeft: pxToRem(30),
    }},
}))
```
accordion stardust

Comment (priyankar205) [reply src/components/Accordion/Accordion.tsx:201]: We will not hard code the styles in the component. We will do it when consuming it.

**Comment by jurokapsiar** [comment]
Please beware that ARIA does not define nested accordions, so coming up with an implementation that works well for all accessibility aspects can be tricky. If this should be used to create the nested list structure, please follow the HTML structure that we have agreed upon and was validated by SMEs.

File: docs/src/examples/components/Accordion/Variations/AccordionNestedExample.shorthand.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    key: 'd',
+    title: 'What is a dimension of a point?',
+    content: 'We need to copy the primary USB firewall!',
+  },
+  {
+    key: 'e',
+    title: 'Nested Accordion',
+    accordion: {
```
Comment (kuzhelov) [inline docs/src/examples/components/Accordion/Variations/AccordionNestedExample.shorthand.tsx:18]: do we really need to support this prop? Would the following shorthand structure work for us:

```js
const nestedPanels = [
  { key: ..., ... }
  ...
]

panels = [
  { key: ..., ... },
  { key: ..., content: <Accordion panels={nestedPanels} ... /> }
]
```

**Comment by levithomason** [comment]
@priyankar205 Just finished stepping through the accessibility details of a Tree vs an Accordion.  Apologies for the change yet again, however, it appears we'll need to accomplish this as a Tree component.  Please sync up with @jurokapsiar for more info.  He and I have already discussed and agreed on a new Tree component.

**Comment by jurokapsiar** [comment]
closing this PR as it has been superseeded by #479

**Pull request closed without merging**